### ISSUE TITLE
Cityline bugs

### ISSUE THREAD

**Issue by yixiao-huang** [max32002/tixcraft_bot#42]
Found two problems in Cityline.
1. There's a bug in the ```cityline_main``` function 
```
    if 'msg.cityline.com' or 'event.cityline.com' in url:
        try:
            WebDriverWait(driver, 10, poll_frequency=0.01).until(
                EC.element_to_be_clickable((By.XPATH, "//button[@id='btn-retry-en-1']"))
            )
            driver.execute_script("goEvent();")
        except Exception as exec1:
            pass
        pass
```
The statement will always be true since the boolean value for the first string is true. You need to add a bracket to the strings and ```or``` operator 
```
    if ('msg.cityline.com' or 'event.cityline.com') in url:
        try:
            WebDriverWait(driver, 10, poll_frequency=0.01).until(
                EC.element_to_be_clickable((By.XPATH, "//button[@id='btn-retry-en-1']"))
            )
            driver.execute_script("goEvent();")
        except Exception as exec1:
            pass
        pass
```
2.  The website will detect the bot and redirect the link to msg.cityline.com which finally directs to the homepage instead of the presale page. I don't know whether to slow the poll frequency will solve this problem

**Comment by dadahaha99** [comment]
係, 今朝搶周杰倫完全唔成功, 只去到主頁, 去不到購買頁面

**Comment by max32002** [comment]
problem1, fixed, but bot detect, I don't know to fix, too.

**Comment by aka8803** [comment]
> problem1, fixed [08d4846](https://github.com/max32002/tixcraft_bot/commit/08d4846b35db64111b26ca9258535b3c4f7e7345), but bot detect, I don't know to fix, too.

any detail about cityline bot detectio?

**Comment by max32002** [comment]
use https://github.com/ultrafunkamsterdam/undetected-chromedriver/
or chrome extension: https://youtu.be/rRl_EETv_b0
